exercises/hexadecimal/hexadecimal_test.go

```diff
@@ -34,18 +43,33 @@ var testCases = []struct {
 
 func TestParseHex(t *testing.T) {
 	for _, test := range testCases {
-		switch out, err := ParseHex(test.in); {
-		case err != nil:
-			if test.errCase == "none" {
+		out, err := ParseHex(test.in)
+		if test.errCase != "none" {
+			// check if err is of error type
+			var _ error = err
+
+			// we expect error
+			if err == nil {
+				t.Errorf("ParseHex(%q): expected an error, but error is nil",
+					test.in)
+			}
+
+			if !strings.Contains(strings.ToLower(err.Error()), test.errCase) {
```

Comment:
if `err == nil` (student's function didn't return an error when it should have), this calls `.Error()` on `nil` and test will crash. I think the solution is to add a `continue` inside the `if err == nil`, does that seem right to you?